### ISSUE TITLE
Add business rule for user name uniqueness.

### ISSUE THREAD

**Issue by benanhalt** [specify/specify7#2260]
So that the following IntegrityError would have been caught as a BusinessRuleException:
```
specifycloud-invertebratepaleontology-1  | [30/Sep/2022 13:22:53] [ERROR] [django.request:222] Internal Server Error: /api/specify/specifyuser/
specifycloud-invertebratepaleontology-1  | Traceback (most recent call last):
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
specifycloud-invertebratepaleontology-1  |     return self.cursor.execute(sql, params)
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 71, in execute
specifycloud-invertebratepaleontology-1  |     return self.cursor.execute(query, args)
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/MySQLdb/cursors.py", line 250, in execute
specifycloud-invertebratepaleontology-1  |     self.errorhandler(self, exc, value)
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler
specifycloud-invertebratepaleontology-1  |     raise errorvalue
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/MySQLdb/cursors.py", line 247, in execute
specifycloud-invertebratepaleontology-1  |     res = self._query(query)
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/MySQLdb/cursors.py", line 412, in _query
specifycloud-invertebratepaleontology-1  |     rowcount = self._do_query(q)
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/MySQLdb/cursors.py", line 375, in _do_query
specifycloud-invertebratepaleontology-1  |     db.query(q)
specifycloud-invertebratepaleontology-1  |   File "/opt/specify7/ve/lib/python3.8/site-packages/MySQLdb/connections.py", line 276, in query
specifycloud-invertebratepaleontology-1  |     _mysql.connection.query(self, query)
specifycloud-invertebratepaleontology-1  | _mysql_exceptions.IntegrityError: (1062, "Duplicate entry 'amaya' for key 'Name'")
specifycloud-invertebratepaleontology-1  | 

```

**Comment by melton-jason** [comment]
This is currently fixed in #2965 (along with other missing uniqueness rules) and will be included with all of the other Business Rule refactoring changes.

**Comment by grantfitzsimmons** [comment]
Fixed in https://github.com/specify/specify7/pull/2965

Will be included in 7.9
